doc/neps/nep-0036-fair-play.rst

```diff
@@ -0,0 +1,180 @@
+==================
+NEP 36 — Fair Play
+==================
+
+:Author: Stéfan van der Walt <stefanv@berkeley.edu>
+:Status: Draft
+:Type: Informational
+:Created: 2019-10-24
+:Resolution: Draft
+
+
+Abstract
+--------
+
+This document describes Rules of Play for engaging with the NumPy
+project as an external entity.  It discusses usage of the name NumPy,
+modified distributions of NumPy, and how to best communicate
+with the project when patched versions are deployed.
+
+Companies and developers will know after reading this NEP what kinds
+of behavior the community would like to see, and which we consider
+troublesome, bothersome, and unacceptable.
+
+Motivation
+----------
+
+It has come to our attention that modified versions of NumPy are
+distributed by external entities.  Sometimes, the patches are
+innocuous, such as when they allow NumPy to function on the target
+system.  In other cases, they add new functionality which causes
+problems for the community.
+
+For example, in December 2018, through the filing of `Issue #12515
+<https://github.com/numpy/numpy/issues/12515>`__, it became known to
+the NumPy project that an external company had published a version of
+NumPy with an extended footprint.  In this specific instance, a bug
+was filed against the `np.erf` function---a function that NumPy itself
+had never provided.
+
+In the months that followed, several conversations were had with the
+entity involved, and a better understanding emerged of needs on either
+side.  The entity agreed to publish the patches made against NumPy,
+and a special label was added on the NumPy issue tracker to identify
+issues pertaining to this specific distribution.
+
+In another example, a security issue (CVE-2019-6446) was filed against
+NumPy.  This forced many distributions to take action, most often by
+changing a default keyword value, unfortunately again leading to
+multiple versions of the NumPy API being deployed, varying across
+distribution.
+
+These are only two of many examples of such modified distributions,
+which cause wasted time for our developers: identifying the release
+from which the error came, how the release was modified from the
+version we provide, and whether we are able to address related issues
+(if not, where to relay the report to).
+
+During a community call on `October 16th, 2019 
+<https://github.com/numpy/archive/blob/master/status_meetings/status-2019-10-16.md>`__
+the community decided to draft a
+set of guidelines for companies and external developers to use when
+they modify and redistribute NumPy.
+
+Scope
+-----
+
+This document aims to define a minimal set of rules that, when
+followed, will placate the NumPy developers.  Our hope, however, is
+that developers who think that they need to modify NumPy will consider
+contributing to the project, or using one of the many mechanisms we
+have for allowing our APIs to operate on externally defined objects.
```

Comment:
```suggestion
There are preferred alternatives to altering our distribution: contributing to the project, or using one of the many mechanisms we
have for allowing our APIs to operate on externally defined objects. This document sets down minimal rules to be followed when these approaches do not suffice.
```

Reply:
This changes what I had intended to say, so leaving out for now.